c10/core/DispatchKey.h

```diff
@@ -437,6 +440,7 @@ enum class DispatchKey : uint16_t {
   // [Masquerading as CUDA]
   _SparseXLA,
   _SparseMLC,
+  _SparseIPU,
```

Comment:
Also on the leading underscore: this is fine, the idea is that by making IPU a "backend component", you're automatically adding space in the dispatcher to be able to register dense + sparse + quantized + autograd kernels. If the Sparse and Quantized keys aren't actually needed yet though (like some of the other keys below), then they have a leading underscore just to make that clear.